aiida/engine/processes/ports.py

```diff
@@ -99,6 +101,22 @@ class InputPort(WithSerialize, WithNonDb, ports.InputPort):
     value serialization to database storable types and support non database storable input types as well.
     """
 
+    def __init__(self, *args, **kwargs):
+        """Override the constructor to check the type of the default if set and warn if not immutable."""
+        # pylint: disable=redefined-builtin,too-many-arguments
+        immutabe_types = (int, float, complex, tuple, frozenset, bytes, six.string_types)
+
+        if 'default' in kwargs:
+            default = kwargs['default']
+            # If the default is specified and it is not a lambda (callable) nor an immutable type, raise a warning.
+            # This is to try and prevent that people set node instances as defaults.
+            if default is not ports.UNSPECIFIED and not (callable(default) or isinstance(default, immutabe_types)):
```

Comment:
btw, all the builtin types are callables, so this can be simplified to:

```python
             if default is not ports.UNSPECIFIED and not callable(default): 
```

Reply:
but `six.string_types` isn't ;)